### PR TITLE
Code Quality: Clean up Widgets

### LINKED ISSUE #13210: Code Quality: Refactor Widget controls
### Description

As #4180 lists this as an issue, widget controls need to be refactored.

Detailed issues:
- Home page is just wrapping WidgetsListControl.
- Each widget control is handling its all functions
- Drive widget won't be updated when drive info is updated 

Keep in mind:
- #5135
- #5136
- #4262
- #6182

### Concerned code

- UserControls.Widgets
- ViewModels.Widgets
- HomeViewModel
- HomePage

### Gains

- A better readability & maintainability.

### Requirements

- Refactor all concened code

### Comments

_No response_

### PR DESCRIPTION
**Pull request by 0x5bfa**
## Summary

As the title.

## Task Checklist

- [x] Clean up Widgets
- [x] Clean up
- [ ] Apply requested changes

## Known Issues

_N/A_

## PR Checklist

- [x] **Closes**: Closes #13210
- [x] **Approval**: Have discussed with and got approval from the team[^1]
- [x] **Tests**: Tested accessibility on the local end
- [x] **Deployment**: Deployed the app on the local end
- [ ] **Localization**: Modified en-US string resources[^2]
- [ ] **Co-authors**: _N/A_

## Steps To Validate Changes

1. Launch the app
2. Go to ...

## Screenshots

_N/A_

[^1]: If the request hasn't been agreed by the team, this work might be rejected. Make sure that you get approval from the team before opening any PR related the request.
[^2]: If you removed any en-US string resources, make sure that there are no references of those resources. When you add a new en-US string resources, do not make any changes on other languages' resources; [Crowdin](https://crowdin.com/project/files-app) will do that, instead.

### CODE REVIEW

**Comment by 0x5bfa** [comment]
Ready for review.

File: src/Files.App/Data/Items/FolderCardItem.cs
Comment (yaira2) [inline src/Files.App/Data/Items/FolderCardItem.cs:1]: What's a folder card item?

File: src/Files.App/Data/Items/WidgetCardItem.cs
Comment (yaira2) [inline src/Files.App/Data/Items/WidgetCardItem.cs:1]: What's a widget card item?

File: src/Files.App/Data/Items/FolderCardItem.cs
Comment (0x5bfa) [reply src/Files.App/Data/Items/FolderCardItem.cs:1]: Used by DriveCardWidget, FolderCardWidget. Base/abstract class of them.

File: src/Files.App/Data/Items/WidgetCardItem.cs
Comment (0x5bfa) [reply src/Files.App/Data/Items/WidgetCardItem.cs:1]: Used by DriveCardWidget, FolderCardWidget. Base/abstract class of them.

Comment (yaira2) [reply src/Files.App/Data/Items/WidgetCardItem.cs:1]: How is different than FolderCardItem?

Comment (0x5bfa) [reply src/Files.App/Data/Items/WidgetCardItem.cs:1]: FolderCardItem derives this class to have path and name implicitly.

Comment (0x5bfa) [reply src/Files.App/Data/Items/WidgetCardItem.cs:1]: I'm gonna add xml comments on them.

File: src/Files.App/Data/Items/FolderCardItem.cs
Comment (0x5bfa) [reply src/Files.App/Data/Items/FolderCardItem.cs:1]: Added comments

**Pull request closed without merging**